crypt-des-obsolete.c

```diff
@@ -152,24 +186,49 @@ SYMVER_encrypt_r;
    these do not get their own file because they're not compiled
    into the static library anyway.  */
 
-#if INCLUDE_setkey || INCLUDE_encrypt
+#if (INCLUDE_setkey || INCLUDE_encrypt) && !ENABLE_OBSOLETE_API_ENOSYS
 static struct des_ctx nr_encrypt_ctx;
 #endif
 
 #if INCLUDE_setkey
+#if ENABLE_OBSOLETE_API_ENOSYS
+void
+setkey (ARG_UNUSED (const char *key))
+{
+  /* Pretend this function is not supported.  */
+  errno = ENOSYS;
+}
+#else
 void
 setkey (const char *key)
 {
   do_setkey_r (key, &nr_encrypt_ctx);
 }
+#endif
 SYMVER_setkey;
 #endif
 
 #if INCLUDE_encrypt
+#if ENABLE_OBSOLETE_API_ENOSYS
+void
+encrypt (char *block, ARG_UNUSED (int edflag))
+{
+  /* Make sure sensitive data is at least zero'ed in
+     case get_random_bytes() fails.  */
```

Comment:
Grammar nit: in formal register `zero'ed` is incorrect.  I would write this comment "Make sure sensitive data is erased in case get_random_bytes fails."

Reply:
Fixed.